stories/core/Table.stories.tsx

```diff
@@ -2117,3 +2122,69 @@ HorizontalScroll.decorators = [
     </div>
   ),
 ];
+
+export const Virtualized: Story<Partial<TableProps>> = (args) => {
+  const onClickHandler = (
+    props: CellProps<{ name: string; description: string }>,
+  ) => action(props.row.original.name)();
+
+  const columns = useMemo(
+    () => [
+      {
+        Header: 'Table',
+        columns: [
+          {
+            id: 'name',
+            Header: 'Name',
+            accessor: 'name',
+          },
+          {
+            id: 'description',
+            Header: 'Description',
+            accessor: 'description',
+          },
+          {
+            id: 'click-me',
+            Header: 'Click',
+            width: 100,
+            Cell: (props: CellProps<{ name: string; description: string }>) => {
+              const onClick = () => onClickHandler(props);
+              return (
+                <a className='iui-anchor' onClick={onClick}>
+                  Click me!
+                </a>
+              );
+            },
+          },
+        ],
+      },
+    ],
+    [],
+  );
+
+  const generateData = useCallback((start: number, end: number) => {
+    const arr = new Array(end - start);
+    for (let i = 0; i < end - start; ++i) {
+      arr[i] = {
+        name: `Name${start + i}`,
+        description: `Description${start + i}`,
+      };
+    }
+    return arr;
+  }, []);
+
+  return (
+    <Table
+      useVirtualization
+      columns={columns}
+      emptyTableContent='No data.'
+      {...args}
+      style={{ height: 440 }}
+      data={generateData(0, 100000)}
```

Comment:
If this virtualization solution produces a noticeable slowdown on first render, I expect it to be unusable with Tree component which could easily contain 10x more rows. Tree component would also need to re-walk its hierarchy and create a new flat list on each data update (e.g. node expansion or collapse).

I somehow doubt that the slowdown can be directly attributed to the data generation procedure. In my Firefox console I can execute it for million elements in under 70ms.

Reply:
The problem might also be `react-table` itself as it is handling each data entry and preparing row and table state.

Reply:
@roluk yup, after all it was an issue with children. I assume they are always being rendered before applying the parent comopnent or something... Anyway, changed API now so it is using render function for the item. If you could check it up now, would appreciate.
Story has a little of lag on react-table (not comprable to previous), but it works nice with just plain divs list (so there might be something with react-table still, but I think minor for now).

Reply:
You can resolve this comment now (the button to do so isn't there for me).